kp_nodes.module

```diff
@@ -504,3 +504,133 @@ function kp_project_germplasm_list($project = null) {
     }
   }
 }
+
+/**
+ * Implements hook_alter().
+ * Match stock name to stock with AGL prefix in rawphenotypes module.
+ * Note: only when saving data in Stage 3.
+ *
+ * @param $data
+ *   An array containing the stock name and project name.
+ *
+ * @return
+ *   A string, stock name with AGL as required.
+ *
+ * NOTE: This applies to check stock names in stage 3: save spreadsheet.
+ */
+function kp_nodes_rawpheno_AGILE_stock_name_alter(&$stock_name, &$project_name) {
+  if ($project_name == 'AGILE: Application of Genomic Innovation in the Lentil Economy') {
+
+    $AGILE_stock_token = 'AGL';
+
+    // Only applies when project is AGILE.
+    if (strpos($stock_name, $AGILE_stock_token) === FALSE) {
+      // Stock name does not contain AGL.
+      // Alter stockname to contain AGL.
+
+      // Space required!
+      $stock_name .= ' AGL';
+    }
+  }
+}
+
+
+/**
+ * Implements hook_alter().
+ * Add support email to rawphenotypes block.
+ *
+ * @param $support_email
+ *   A string, email address.
+ *
+ * @return
+ *   A string, email address.
+ *
+ * NOTE: to set this value, run/execute this line in Drupal /devel/php
+ *       variable_set('kp_rawpheno_support_email', 'your-email@email.com')
+ *       otherwise, the module will load email from either the Drupal user 0 
+ *       or the Drupal site wide mail.
+ */
+function kp_nodes_rawpheno_support_email_alter(&$support_email) {
+  $support_email = variable_get('kp_rawpheno_support_email');
+  
+  if (empty($support_email)) {
+    // Get Drupal admin 0 email.
+    $support_email = chado_query("SELECT mail FROM users WHERE uid = 1")
+      ->fetchField();
+    
+    if (empty($support_email)) {
+      // Get the Drupal setup site email address.
+      $support_email = variable_get('site_mail');
+    }
+  }
+}
+
+
+/**
+ * Implements hook_alter().
+ * Ignore columns in a spreadsheet file.
+ *
+ * @param $columns
+ *   An array, as set of column headers from a spreadsheet file.
+ *
+ * @return
+ *   An array, the same column headers in the parameter but with the
+ *   specified column(s) removed.
+ */
+
+// When searching for new columns.
+// in rawpheno.upload.form.inc : rawpheno_upload_form_stage_review().

```

Comment:
Please be more descriptive in your docblocks. Also there should not be any code between the docblock and the function header. The kp_nodes_rawpheno_support_email_alter() docblock was perfect :-) so just keep with that format for All Alter functions while substituting in the correct info (e.g. columns, etc.).